panther_analysis_tool/rule.py

```diff
@@ -133,6 +139,18 @@ def __init__(self, config: Mapping):
         self.detection_id = config["id"]
         self.detection_version = config["versionId"]
 
+        # backwards compatible for legacy mocking
+        self._setup_exception = None
+        if "lastUpdatedVersion" in config:
+            try:
+                detection_panther_version = semver.VersionInfo.parse(config["lastUpdatedVersion"])
+            except Exception as err:  # pylint: disable=broad-except
+                self._setup_exception = err
+                return
+            self.use_legacy_mocking = detection_panther_version < LEGACY_MOCKING_VERSION
+        else:
+            self.use_legacy_mocking = True
```

Comment:
Is there a reason to set this mocking field in the detection rather than outside the detection? (i.e. set the `detection_panther_version` here and just use that in `_resolve_mocks ` method?)

Reply:
No particular reason, I just figured it'd be more readable if we were to implement the semver comparison here